### PR TITLE
Cu resolver extension point fragment

### PR DESCRIPTION
**Pull request by robstryker**

### CODE REVIEW

**Comment by robstryker** [comment]
This is the incubator version of the patch. A lot of it reverts changes to ASTParser that were made in incubator much earlier.

**Comment by robstryker** [comment]
To see what a jdt-only PR would look like, please see the following:   https://github.com/eclipse-jdt/eclipse.jdt.core/compare/master...robstryker:eclipse.jdt.core:jdt_core_pr_curesolution_extpt?expand=1

File: org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ICompilationUnitResolver.java [hunk 1]
```diff
@@ -13,44 +13,93 @@
...
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.jdt.core.ICompilationUnit;
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.jdt.core.WorkingCopyOwner;
 import org.eclipse.jdt.internal.compiler.batch.FileSystem.Classpath;
 
-interface ICompilationUnitResolver {
-
+/**
+ * @since 3.38
```
Comment (mickaelistria) [inline org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ICompilationUnitResolver.java:26]: Please add a bit of documentation here.

**Review by mickaelistria — COMMENTED** [review]
Once the documentation is added, I think we can just merge it.

**Pull request merged**